src/App.tsx

```diff
@@ -1,26 +1,361 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo, Filter, Erors } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components';
 
 export const App: React.FC = () => {
+  const input = useRef<HTMLInputElement>(null);
+  const hideErrorTimer = useRef<number | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number[] | null>(null);
+  const [filter, setFilter] = useState(Filter.All);
+  const [title, setTitle] = useState('');
+  const [loadingError, setLoadingError] = useState({
+    queryError: false,
+    addError: false,
+    todosError: false,
+    deleteError: false,
+    updateError: false,
+  });
+
+  const anyExistingError =
+    loadingError.queryError ||
+    loadingError.addError ||
+    loadingError.todosError ||
+    loadingError.deleteError ||
+    loadingError.updateError;
+
+  const focusInput = () => (input.current ? input.current.focus() : null);
+
+  const clearError = useCallback(() => {
+    if (hideErrorTimer.current) {
+      clearTimeout(hideErrorTimer.current);
+      hideErrorTimer.current = null;
+    }
+
+    setLoadingError({
+      queryError: false,
+      addError: false,
+      todosError: false,
+      deleteError: false,
+      updateError: false,
+    });
+  }, []);
+
+  const showError = useCallback(
+    (errorType: Erors) => {
```

Comment:
```suggestion
    (errorType: Errors) => {
```
use singular form for types